lib/utils/remote-build.ts

```diff
@@ -110,68 +109,76 @@ async function getBuilderEndpoint(
 }
 
 export async function startRemoteBuild(build: RemoteBuild): Promise<void> {
-	const stream = await getRemoteBuildStream(build);
+	const [buildRequest, stream] = await getRemoteBuildStream(build);
+
+	// Setup CTRL-C handler so the user can interrupt the build
+	let cancellationPromise = Promise.resolve();
+	const sigintHandler = () => {
+		process.exitCode = 130;
+		console.error('\nReceived SIGINT, cleaning up. Please wait.');
+		try {
+			cancellationPromise = cancelBuildIfNecessary(build);
+		} catch (err) {
+			console.error(err.message);
+		} finally {
+			buildRequest.abort();
+			const sigintErr = new SIGINTError('Build aborted on SIGINT signal');
+			sigintErr.code = 'SIGINT';
+			stream.emit('error', sigintErr);
+		}
+	};
 
-	// Special windows handling (win64 also reports win32)
-	if (process.platform === 'win32') {
-		const rl = readline.createInterface({
-			input: process.stdin,
-			output: process.stdout,
-		});
+	const { addSIGINTHandler } = await import('./helpers');
+	addSIGINTHandler(sigintHandler);
 
-		rl.on('SIGINT', () => process.emit('SIGINT' as any));
+	try {
+		if (build.opts.headless) {
+			await handleHeadlessBuildStream(stream);
+		} else {
+			await handleRemoteBuildStream(build, stream);
+		}
+	} finally {
+		process.removeListener('SIGINT', sigintHandler);
+		globalLogger.outputDeferredMessages();
+		await cancellationPromise;
```

Comment:
By the way, during testing I came across issue #2121, but it was pre-existing and the fix lies in the balena-builder repo I believe.